src/linux.rs

```diff
@@ -34,10 +36,84 @@ fn make_netlink_message(ifaddrmsg: NlPayload<Ifaddrmsg>) -> Nlmsghdr<Rtm, NlPayl
     )
 }
 
-/// Retrieves the local IP address fo this system
+/// Retrieves the local IP address for this system
 pub fn local_ip() -> Result<IpAddr, Error> {
     let mut netlink_socket = NlSocketHandle::connect(NlFamily::Route, None, &[])
         .map_err(|err| Error::StrategyError(err.to_string()))?;
+
+    let dstip = Ipv4Addr::new(192, 0, 2, 0); // reserved external IP
+    let rawdstip = u32::from(dstip).to_be();
+    let routeattr = Rtattr::new(None, Rta::Dst, rawdstip)
+        .map_err(|err| Error::StrategyError(err.to_string()))?;
+    let mut routepayload = RtBuffer::new();
+    routepayload.push(routeattr);
```

Comment:
Nitpick but can we split both words to read them with ease?

```suggestion
    let mut route_payload = RtBuffer::new();
    route_payload.push(routeattr);
```